File: app/schemas/sensor.py
# app/schemas/sensor.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime
from enum import Enum

class AlertLevel(str, Enum):
    INFO = "info"
    WARNING = "warning"
    CRITICAL = "critical"

class AlertStatus(str, Enum):
    NEW = "new"
    ONGOING = "ongoing"
    RESOLVED = "resolved"

# Base schemas
class SensorBase(BaseModel):
    name: str
    description: Optional[str] = None
    metric_key: str
    alert_condition: str
    alert_level: AlertLevel = AlertLevel.WARNING

# Create schema
class SensorCreate(SensorBase):
    device_id: int

# Update schema
class SensorUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    metric_key: Optional[str] = None
    alert_condition: Optional[str] = None
    alert_level: Optional[AlertLevel] = None
    is_active: Optional[bool] = None

# Alert schemas
class AlertBase(BaseModel):
    value: float
    message: str
    is_resolved: bool = False

class AlertCreate(AlertBase):
    sensor_id: int

class AlertResponse(AlertBase):
    id: int
    sensor_id: int
    timestamp: datetime
    status: AlertStatus
    first_detected_at: datetime
    last_checked_at: datetime
    resolution_time: Optional[datetime] = None
    consecutive_checks: int = 1

    class Config:
        from_attributes = True

# Response schema
class SensorResponse(SensorBase):
    id: int
    device_id: int
    created_at: datetime
    updated_at: datetime
    is_active: bool
    alerts: List[AlertResponse] = []

    class Config:
        from_attributes = True

File: infra-platform-frontend/src/types/sensor.ts
// src/types/sensor.ts

export enum AlertLevel {
  INFO = "info",
  WARNING = "warning",
  CRITICAL = "critical"
}

export enum AlertStatus {
  NEW = "new",
  ONGOING = "ongoing",
  RESOLVED = "resolved"
}

export interface Alert {
  id: number;
  sensor_id: number;
  value: number;
  message: string;
  is_resolved: boolean;
  timestamp: string;
  status: AlertStatus;
  first_detected_at: string;
  last_checked_at: string;
  resolution_time: string | null;
  consecutive_checks: number;
}

export interface Sensor {
  id: number;
  name: string;
  description: string | null;
  device_id: number;
  metric_key: string;
  alert_condition: string;
  alert_level: AlertLevel;
  created_at: string;
  updated_at: string;
  is_active: boolean;
  alerts: Alert[];
}

export interface SensorCreate {
  name: string;
  description?: string;
  device_id: number;
  metric_key: string;
  alert_condition: string;
  alert_level?: AlertLevel;
}

export interface SensorUpdate {
  name?: string;
  description?: string;
  metric_key?: string;
  alert_condition?: string;
  alert_level?: AlertLevel;
  is_active?: boolean;
}

export interface AlertCreate {
  sensor_id: number;
  value: number;
  message: string;
  is_resolved?: boolean;
}

export interface AlertFilter {
  device_id?: number;
  days?: number;
}